kratos/tests/cpp_tests/factories/test_strategy_factory.cpp

```diff
@@ -0,0 +1,172 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "containers/model.h"
+
+// Utility includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "factories/base_factory.h"
+#include "spaces/ublas_space.h"
+
+namespace Kratos
+{
+    namespace Testing
+    {
+        /// Tests
+
+        // Spaces
+        using SparseSpaceType = UblasSpace<double, CompressedMatrix, Vector>;
+        using LocalSpaceType = UblasSpace<double, Matrix, Vector>;
+        using LinearSolverType = LinearSolver<SparseSpaceType, LocalSpaceType>;
+
+        /// The definition of the strategy
+        using StrategyType = SolvingStrategy<SparseSpaceType,LocalSpaceType, LinearSolverType>;
+
+        /// The definition of the factory
+        using StrategyFactoryType = BaseFactory<StrategyType>;
+
+        /**
+         * Checks if the ResidualBasedLinearStrategy performs correctly the Factory
+         */
+        KRATOS_TEST_CASE_IN_SUITE(ResidualBasedLinearStrategyFactory, KratosCoreFastSuite)
+        {
+            Model this_model;
+            auto& r_model_part = this_model.CreateModelPart("Main");
+            Parameters this_parameters = Parameters(R"({"name" : "linear_strategy",
+                                                        "linear_solver_settings" : {
+                                                            "solver_type" : "amgcl"
+                                                        },
+                                                        "scheme_settings" : {
+                                                            "name" : "static"
+                                                        },
+                                                        "convergence_criteria_settings" : {
+                                                            "name" : "displacement_criteria"
+                                                        },
+                                                        "builder_and_solver_settings" : {
+                                                            "name" : "elimination_builder_and_solver"
+                                                        }
+                                                        })");
+            StrategyType::Pointer p_strategy = StrategyFactoryType().Create(r_model_part, this_parameters);
+            KRATOS_CHECK_C_STRING_EQUAL((p_strategy->Info()).c_str(), "ResidualBasedLinearStrategy");
+        }
```

Comment:
Please correct me if I am wrong in saying the following : 
In python level, when one makes the strategy in the `solver.py`  one should make a similar looking `this_parameters` and pass it into the factory so the strategy object is made. This means one need not construct explicitly the objects of scheme, b&s and solver. 

This also means that there is no need to expose those (scheme, b&s and solver) to the python level correct ?

Reply:
?

Reply:
I do not understand

Reply:
Ok .. 
how are these factories intended to be used from python ? Is there an example ? Any test ?

Reply:
It is the same procedure than the linear solver factories. Do you need a test in python?

Reply:
Yes Please add a test for Python !

Reply:
For all of them, especially strategy factory !

Reply:
Okay

Reply:
Done